### ISSUE TITLE
Broken Markdown in gtk::Text documentation

### ISSUE THREAD

**Issue by AaronErhardt** [gtk-rs/gtk4-rs#521]
Link to docs: [https://gtk-rs.org/gtk4-rs/git/docs/gtk4/struct.Text.html](https://gtk-rs.org/gtk4-rs/git/docs/gtk4/struct.Text.html)

This is the text as it starts in the docs:

```markdown
The [`Text`][crate::Text] widget is a single-line text entry widget.

[`Text`][crate::Text] is the common implementation of single-line text editing
that is shared between [`Entry`][crate::Entry], `GtkPasswordEntry, [`SpinButton`][crate::SpinButton]
and other widgets. In all of these, [`Text`][crate::Text] is used as the delegate
for the [`Editable`][crate::Editable] implementation.
```

As you can see there's a missing backticks after GtkPasswordEntry that causes the following text to be rendered incorrectly. I think that ```[`PasswordEntry`][crate::PasswordEntry]``` was intended to be used here but was accidentally overlooked.

The doc links the source at [https://gtk-rs.org/gtk4-rs/git/docs/src/gtk4/auto/text.rs.html](https://gtk-rs.org/gtk4-rs/git/docs/src/gtk4/auto/text.rs.html) but I looked at `gtk4/src/auto/text.rs` and couldn't find these doc comments so they must be somewhere else. Also this file was probably generated automatically by gir.

**Comment by bilelmoussaoui** [comment]
The docs are added automatically on CI from the gobject introspection data with gir. You can run `python3 generator.py --embed-docs ./gtk4` to add the docs to your local files if you want to look at it. The issue here is how we replace the C symbols by rust ones  and the rust regex crate doesn't help much.

**Comment by bilelmoussaoui** [comment]
After a bit of investigation, this seems to be an issue in the upstream docs and was fixed in https://gitlab.gnome.org/GNOME/gtk/-/commit/3ab34ffdc09b2cd2c29b4e1661ebb3ed5395ed27. It should be fixed once #505 lands
